### DISCUSSION TITLE
filterable() and Status fields

### DISCUSSION THREAD

**Discussion by tommiekn** [laravel/nova-issues#5032] [Q&A] [answered]
- Laravel Version: 9.38.0
- Nova Version: 4.18.2
- PHP Version: 8.1.11

### Description:
filterable() doesn't work with Status fields.

### Detailed steps to reproduce the issue on a fresh Nova installation:
While using filterable() method in a **Status** field, it returns the error:
"$filterable needs to be callable/closure"

Haven't found anything about this issue, either in the docs. For my understanding filterable() should work on any kind of field.

```
Status::make('Status')
            ->loadingWhen(['pending',])
            ->failedWhen(['failed',])->filterable(),
```

Is still possible to create a custom filter and use that to filter it, that works. Only the filterable() doesnt work.

**Comment by crynobone** [comment, marked as answer]
![image](https://user-images.githubusercontent.com/172966/200427531-9ead7bd7-e808-4c82-99dc-52a6b1497d6d.png)

The error is correct. Since we can't 100% verify the correct query to be used for the filter you need to manually handle it.

**Reply by tommiekn** [reply]
Hello, will try this way thank you.
Fyi, not sure i have looked in the wrong place but I haven't found any docs on the closure argument, maybe should be added:
https://nova.laravel.com/docs/4.0/releases.html#filterable-fields
